server/app.js

```diff
@@ -1,13 +1,66 @@
 const express = require('express');
 const cors = require('cors');
+const passport = require('passport');
+const GoogleStrategy = require('passport-google-oauth').OAuth2Strategy;
 const db = require('./db-connection');
 require('dotenv').config();
 
+const corsOptions = {
+  origin: 'http://localhost:3000',
+  credentials: true,
+  optionsSuccessStatus: 200, // some legacy browsers (IE11, various SmartTVs) choke on 204
+};
+
 const app = express();
 app.use(express.json());
-app.use(cors());
+
+app.use(cors(corsOptions));
+app.use(passport.initialize());
+app.use(passport.session());
 const port = 8000;
 
+const GOOGLE_CLIENT_ID = process.env.GOOGLECLIENT;
+const GOOGLE_CLIENT_SECRET = process.env.GOOGLESECRET;
+
+passport.use(
+  new GoogleStrategy(
+    {
+      clientID: GOOGLE_CLIENT_ID,
+      clientSecret: GOOGLE_CLIENT_SECRET,
+      callbackURL: 'http://localhost:8000/auth/google/callback',
+    },
+    async (accessToken, refreshToken, profile, done) => {
+      // QUERY FOR USER!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+      const query = await db.query(
+        'SELECT email FROM users',
+      );
+      for (let i = 0; i < query.rows.length; i += 1) {
+        if (query.rows[i].email === profile.emails[0].value) {
+          return done(null, profile);
+        }
+      }
+      return done(null, false);
+    },
+  ),
+);
+
+passport.serializeUser((user, done) => {
+  done(null, user.id);
+});
+
+passport.deserializeUser((user, done) => {
+  done(null, user.displayName);
+});
```

Comment:
deserializeUser should take in the identifier that you serialize a user into. in our case, `user.id` from
```
passport.serializeUser((user, done) => {
  done(null, user.id);
});
```
Afterwards, deserializeUser should output the original user object (which in our case would be the row of the user within the `members` table and pass that into `done()`